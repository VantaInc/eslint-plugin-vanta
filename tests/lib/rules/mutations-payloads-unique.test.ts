import { GraphQLRuleTester } from "@graphql-eslint/eslint-plugin";
import rule from "../../../lib/rules/mutations-payloads-unique";

const ruleTester = new GraphQLRuleTester();

ruleTester.runGraphQLTests("mutations-payloads-unique", rule, {
  valid: [
    {
      code: `
type Mutation {
  validMutation: abcPayload
  anotherOne: bcdPayload
}
`,
    },
    {
      code: `
type Mutation {
  validMutation: abcPayload
}

extend type Mutation {
  anotherOne: bcdPayload
}
`,
    },
  ],
  invalid: [
    {
      code: `
type Mutation {
  firstMutation: abcPayload
  secondMutation: abcPayload
}
`,
      errors: 1,
    },
    {
      code: `
type Mutation {
  firstMutation: abcPayload
  secondMutation: bcdPayload
  thirdMutation: abcPayload
}
`,
      errors: 1,
    },
    {
      code: `
    type Mutation {
      firstMutation: abcPayload
    }

    extend type Mutation {
      secondMutation: abcPayload}
    `,
      errors: 1,
    },
    {
      code: `
    type Mutation {
      firstMutation: abcPayload
      secondMutation: bcdPayload
      thirdMutation: abcPayload
    }

    extend type Mutation {
      fourthMutation: bcdPayload
    }
    `,
      errors: 2,
    },
  ],
});
