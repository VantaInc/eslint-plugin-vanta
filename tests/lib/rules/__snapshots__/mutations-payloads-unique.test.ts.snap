// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invalid #1 1`] = `
"#### ⌨️ Code

      1 | type Mutation {
      2 |   firstMutation: abcPayload
      3 |   secondMutation: abcPayload
      4 | }

#### ❌ Error

    > 1 | type Mutation {
        | ^^^^^^^^^^^^^^^
    > 2 |   firstMutation: abcPayload
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 3 |   secondMutation: abcPayload
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 4 | }
        | ^^ Mutation payload types must be unique; reused type abcPayload"
`;

exports[`Invalid #2 1`] = `
"#### ⌨️ Code

      1 | type Mutation {
      2 |   firstMutation: abcPayload
      3 |   secondMutation: bcdPayload
      4 |   thirdMutation: abcPayload
      5 | }

#### ❌ Error

    > 1 | type Mutation {
        | ^^^^^^^^^^^^^^^
    > 2 |   firstMutation: abcPayload
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 3 |   secondMutation: bcdPayload
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 4 |   thirdMutation: abcPayload
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 5 | }
        | ^^ Mutation payload types must be unique; reused type abcPayload"
`;

exports[`Invalid #3 1`] = `
"#### ⌨️ Code

      1 |     type Mutation {
      2 |       firstMutation: abcPayload
      3 |     }
      4 |
      5 |     extend type Mutation {
      6 |       secondMutation: abcPayload}

#### ❌ Error

    > 1 |     type Mutation {
        | ^^^^^^^^^^^^^^^^^^^
    > 2 |       firstMutation: abcPayload
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 3 |     }
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 4 |
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 5 |     extend type Mutation {
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 6 |       secondMutation: abcPayload}
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Mutation payload types must be unique; reused type abcPayload"
`;

exports[`Invalid #4 1`] = `
"#### ⌨️ Code

      1 |     type Mutation {
      2 |       firstMutation: abcPayload
      3 |       secondMutation: bcdPayload
      4 |       thirdMutation: abcPayload
      5 |     }
      6 |
      7 |     extend type Mutation {
      8 |       fourthMutation: bcdPayload
      9 |     }

#### ❌ Error 1/2

    > 1 |     type Mutation {
        | ^^^^^^^^^^^^^^^^^^^
    > 2 |       firstMutation: abcPayload
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 3 |       secondMutation: bcdPayload
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 4 |       thirdMutation: abcPayload
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 5 |     }
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 6 |
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 7 |     extend type Mutation {
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 8 |       fourthMutation: bcdPayload
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 9 |     }
        | ^^^^^^ Mutation payload types must be unique; reused type abcPayload

#### ❌ Error 2/2

    > 1 |     type Mutation {
        | ^^^^^^^^^^^^^^^^^^^
    > 2 |       firstMutation: abcPayload
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 3 |       secondMutation: bcdPayload
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 4 |       thirdMutation: abcPayload
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 5 |     }
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 6 |
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 7 |     extend type Mutation {
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 8 |       fourthMutation: bcdPayload
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 9 |     }
        | ^^^^^^ Mutation payload types must be unique; reused type bcdPayload"
`;
